### DISCUSSION TITLE
Frequently asked questions / issues

### DISCUSSION THREAD

**Discussion by raouldeheer** [justkey007/tsc-alias#110] [General]
## _Replaces path that shouldn't be replaced_

This is probably because it's resolving to the baseUrl. This could be solved by adding a small config to tsconfig.json.
More info in issues: #71 #79 #103 #109 

```json
{
  "compilerOptions": {
    ...
  },
  "tsc-alias": {
    "replacers": {
      "base-url": {
        "enabled": false
      }
    }
  }
}
```

_Why is this enabled by default?_

"All module imports with non-relative names are assumed to be relative to the [baseUrl](https://www.typescriptlang.org/tsconfig#baseUrl)." -[typescript handbook](https://www.typescriptlang.org/docs/handbook/module-resolution.html#base-url) 
Therefore base-url is enabled by default.

## _Do I need to add a replacer?_

No, you don't have to add a replacer. Adding an extra replacer to tsc-alias is optional.

## _Alias isn't replaced when target file isn't present_

This happens because tsc-alias checks if the file exists, because if it doesn't it will use the next path in the array (in the tsconfig each path has an array of solutions).
But this problem can be easily fixed by adding dummy files so that the file exists and resolves.

**Comment by ianyong** [comment]
> This is probably because it's resolving to the baseUrl. This could be solved by adding a small config to tsconfig.json.

For anyone facing this issue, [the `paths` compiler option can be used without the `baseUrl` compiler option since TypeScript 4.1](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#paths-without-baseurl). If you're not using the `baseUrl` compiler option for other purposes, it's probably better to remove the option entirely instead of overriding the `base-url` replacer behaviour as suggested in the FAQ.
